plugins/antispam.py

```diff
@@ -0,0 +1,553 @@
+                self.calc_thresholds(msg.guild)
+                await respond(msg, f"**AFFIRMATIVE. ANALYSIS: applied infraction threshold options:**\n"
+                                   f"```{t_string}```")
+            else:
+                await respond(msg, "**WARNING: No valid arguments detected.**")
+        elif len(args) > 3:
+            if args[1].lower() == "set":
+                # allows user-friendly one-option commands like
+                # !spam_infs set react 1
+                t_arg = args[2].lower()
+                if t_arg in react_strings:
+                    try:
+                        t_cfg["thresholds"][0] = max(1, int(args[3]))
+                    except ValueError:
+                        raise SyntaxError("Expected integer value.")
+                    else:
+                        await respond(msg, f"**AFFIRMATIVE. Spam response now escalates to emoji reaction on "
+                                           f"{ordinal(max(1, int(args[3])))} infraction.**")
+                elif t_arg in del_strings:
+                    try:
+                        t_cfg["thresholds"][1] = max(1, int(args[3]))
+                    except ValueError:
+                        raise SyntaxError("Expected integer value.")
+                    else:
+                        await respond(msg, f"**AFFIRMATIVE. Spam response now escalates to deletion on "
+                                           f"{ordinal(max(1, int(args[3])))} infraction after previous level.**")
+                elif t_arg in role_strings:
+                    try:
+                        t_cfg["thresholds"][2] = max(1, int(args[3]))
+                    except ValueError:
+                        raise SyntaxError("Expected integer value.")
+                    else:
+                        await respond(msg, f"**AFFIRMATIVE. Spam response now escalates to role application on "
+                                           f"{ordinal(max(1, int(args[3])))} infraction after previous level.**")
+                elif t_arg in ban_strings:
+                    try:
+                        t_cfg["thresholds"][3] = max(1, int(args[3]))
+                    except ValueError:
+                        raise SyntaxError("Expected integer value.")
+                    else:
+                        await respond(msg, f"**AFFIRMATIVE. Spam response now escalates to banning on "
+                                           f"{ordinal(max(1, int(args[3])))} infraction after previous level.**")
+                elif t_arg in time_strings:
+                    try:
+                        t_cfg["infraction_timeout"] = max(1, int(args[3]))
+                    except ValueError:
+                        raise SyntaxError("Expected integer value.")
+                    else:
+                        await respond(msg, f"**AFFIRMATIVE. Infraction is now taken off every "
+                                           f"{p_time(max(1, int(args[3])))}.")
```

Comment:
Holy duplicate code! Solution: have the `if...elif` block set the index to use, and use a single `try...except...else` block.